addons/website_sale/models/res_config_settings.py

```diff
@@ -69,31 +69,24 @@ def _compute_terms_url(self):
         for record in self:
             record.terms_url = '%s/terms' % record.website_id.get_base_url()
 
-    @api.model
-    def get_values(self):
-        res = super(ResConfigSettings, self).get_values()
-
-        sale_delivery_settings = 'none'
-        if self.env['ir.module.module'].search([('name', '=', 'delivery')], limit=1).state in ('installed', 'to install', 'to upgrade'):
-            sale_delivery_settings = 'internal'
-            if self.env['ir.module.module'].search([('name', '=', 'website_sale_delivery')], limit=1).state in ('installed', 'to install', 'to upgrade'):
-                sale_delivery_settings = 'website'
-
-        res.update(
-            sale_delivery_settings=sale_delivery_settings,
-            enabled_extra_checkout_step=self.env.ref('website_sale.extra_info_option').active,
-            enabled_buy_now_button=self.env.ref('website_sale.product_buy_now').active,
-        )
-        return res
-
     def set_values(self):
         super().set_values()
-        extra_step_view = self.env.ref('website_sale.extra_info_option')
-        if extra_step_view.active != self.enabled_extra_checkout_step:
-            extra_step_view.active = self.enabled_extra_checkout_step
-        buy_now_view = self.env.ref('website_sale.product_buy_now')
-        if buy_now_view.active != self.enabled_buy_now_button:
-            buy_now_view.active = self.enabled_buy_now_button
+        if self.website_id:
+            extra_step_views = self.with_context(website_id=self.website_id.id).website_id.viewref(
+                'website_sale.extra_info_option')
+            buy_now_views = self.with_context(website_id=self.website_id.id).website_id.viewref(
+                'website_sale.product_buy_now')
+        else:
+            buy_now_views = self.env['ir.ui.view'].search([('key', '=', 'website_sale.product_buy_now')])
+            extra_step_views = self.env['ir.ui.view'].search([('key', '=', 'website_sale.extra_info_option')])
```

Comment:
I never heard we support/want to support editing multiple websites at once by emptying the website field.
I don't think we should do anything when no website is selected in the settings.